### DISCUSSION TITLE
Where to put extensions in the file system so these can be loaded using a relative path?

### DISCUSSION THREAD

**Discussion by mskyttner** [duckdb/duckdb#4046] [Q&A] [answered]
If loading an extension built locally - where should it be located in the file system when using a relative path in duckdb CLI to refer to it (ie if issuing `load 'httpfs.duckdb_extension';` - what paths are searched?)

Are paths searched relative to the home directory of the user running the duckdb process or some other location?

**Comment by Mytherin** [comment, marked as answer]
You can use `select * from duckdb_extensions();` to see a list of installed extensions and their file paths. They are installed into the `~/.duckdb/[githash]/` folder.

**Comment by orzom411** [comment]
I believe you can, might want to try being specific in the syntax, e.g. include a path, relative or full, to the extension.  

load './httpfs.duckdb_extension';

Is how I was testing the httpfs extension compiled locally and it worked well.  Hope that helps.

**Reply by mskyttner** [reply]
It helps, thanks!

**Reply by mskyttner** [reply]
The . in your example, which path does it refer to, would that be the location of the duckdb binary?

**Reply by orzom411** [reply]
The current path you started DuckDB from, not where DuckDB.exe resides.  For example in windows let's say I do the following in at a command prompt:

```
md c:\temp\2022\07\20220708
cd c:\temp\2022\07\20220708
```

Suppose I copy httpfs.duckdb_extension into this path, e.g. I've compiled and pulled it here as this is the day I compiled this version for testing.

Assuming I have extracted the current release distribution into, I'd start DuckDB:

`c:\duckdb\duckdb.exe`

The folder I am in is c:\temp\2022\07\20220708

`load './httpfs.duckdb_extension';`

Is the same as:

`load 'c:/temp/2022/07/20220708/httpfs.duckdb_extension';`

I like this method as I can make adjustments and work through the bugs I've introduced in a somewhat orderly fashion.  Cleanup is also a snap, I just delete the tree and gone is all the testing mess. 

By the way you can also use .shell to see where you are.  For example:

```
v0.4.1-dev17 c2cce5ad8
Enter ".help" for usage hints.
Connected to a transient in-memory database.
Use ".open FILENAME" to reopen on a persistent database.
D .shell 'cd'
D:\Temp\2022\04\20220708
```

Hope that all helps.
